### PR TITLE
OSDOCS-10038:Updated support for transparent forward proxy info

### PR DESCRIPTION
**Pull request by AedinC**
Version(s):
4.15+

Issue:
https://issues.redhat.com/browse/OSDOCS-10038

Link to docs preview:

OSD Docs:
- https://75977--ocpdocs-pr.netlify.app/openshift-dedicated/latest/networking/configuring-cluster-wide-proxy.html
- https://75977--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_install_access_delete_cluster/creating-a-gcp-cluster.html
- https://75977--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_install_access_delete_cluster/creating-an-aws-cluster.html

ROSA Docs:
- https://75977--ocpdocs-pr.netlify.app/openshift-rosa/latest/networking/configuring-cluster-wide-proxy.html
- https://75977--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.html

Peer reviewer:
- [x] Peer reviewer has approved this change.

SME review:
- [x] SME has approved this change.

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by AedinC** [comment]
@ciaranRoche @yingzhanredhat Can you please review whenever you get a chance? Thanks!

**Comment by aldiazRH** [comment]
No changed needed, LGTM!

**Review by michaelryanmcneill — CHANGES_REQUESTED** [review]
This information is incorrect. We shouldn't be telling customers about an install-config.yaml parameter anyways. Specifying "always" doesn't make it disregard the other proxy values, it just includes the certificate if a http and https proxy are not passed. If they are passed, it behaves just like proxypass.

**Comment by AedinC** [comment]
> This information is incorrect. We shouldn't be telling customers about an install-config.yaml parameter anyways. Specifying "always" doesn't make it disregard the other proxy values, it just includes the certificate if a http and https proxy are not passed. If they are passed, it behaves just like proxypass.

Thanks @michaelryanmcneill. I thought we had captured all the docs updates related to support for forward transparent proxies in #74572, but I received PM feedback that we need to mention that we're setting additional trust bundle policy set to Always. I then muddied the water between OCP and ROSA/OSD in an effort to add more context.
Should I remove the info about the additional trust bundle policy, and only add that if the customer is using a transparent proxy they must include the additional-trust-bundle-file?

File: modules/configuring-a-proxy-after-installation-cli.adoc [hunk 1]
```diff
@@ -36,7 +36,7 @@ $ rosa edit cluster \
...
-<2> The `additional-trust-bundle-file` argument is a file path pointing to a bundle of PEM-encoded X.509 certificates, which are all concatenated together. The `additionalTrustBundle` parameter is required unless the identity certificate of the proxy is signed by an authority from the {op-system} trust bundle. If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional CAs, you must provide the MITM CA certificate.
+<2> The `additional-trust-bundle-file` argument is a file path pointing to a bundle of PEM-encoded X.509 certificates, which are all concatenated together. The `additionalTrustBundle` parameter is required unless the identity certificate of the proxy is signed by an authority from the {op-system} trust bundle and also for transparent proxies. If you provide an additional trust bundle, {cluster-manager-first} sets the additional `TrustBundlePolicy` to *Always*. If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional CAs, you must provide the MITM CA certificate.
```
Comment (michaelryanmcneill) [inline modules/configuring-a-proxy-after-installation-cli.adoc:39]: Suggested revision:
> The `additional-trust-bundle-file` argument is a file path pointing to a bundle of PEM-encoded X.509 certificates, which are all concatenated together. The `additional-trust-bundle-file` argument is required for users who use a TLS-inspecting proxy unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle. This applies regardless of whether the proxy is transparent or requires explicit configuration using the `http-proxy` and `https-proxy` arguments.

File: modules/configuring-a-proxy-during-installation-cli.adoc [hunk 2]
```diff
@@ -30,7 +30,7 @@ $ rosa create cluster \
...
-<2> The `additional-trust-bundle-file` argument is a file path pointing to a bundle of PEM-encoded X.509 certificates, which are all concatenated together. The `additionalTrustBundle` parameter is required unless the identity certificate of the proxy is signed by an authority from the {op-system} trust bundle. If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional CAs, you must provide the MITM CA certificate.
+<2> The `additional-trust-bundle-file` argument is a file path pointing to a bundle of PEM-encoded X.509 certificates, which are all concatenated together. The `additionalTrustBundle` parameter is required unless the identity certificate of the proxy is signed by an authority from the {op-system} trust bundle and also for transparent proxies. If you provide an additional trust bundle, {cluster-manager-first} sets the additional `TrustBundlePolicy` to *Always*. If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional CAs, you must provide the MITM CA certificate.
```
Comment (michaelryanmcneill) [inline modules/configuring-a-proxy-during-installation-cli.adoc:33]: Suggested revision:
> The `additional-trust-bundle-file` argument is a file path pointing to a bundle of PEM-encoded X.509 certificates, which are all concatenated together. The `additional-trust-bundle-file` argument is required for users who use a TLS-inspecting proxy unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle. This applies regardless of whether the proxy is transparent or requires explicit configuration using the `http-proxy` and `https-proxy` arguments.

File: modules/configuring-a-proxy-after-installation-ocm.adoc [hunk 3]
```diff
@@ -41,6 +41,9 @@ endif::openshift-dedicated[]
 +
 If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional certificate authorities (CAs), you must provide the MITM CA certificate.
 +
+--
+include::snippets/rosa-add-trust-bundle-policy.adoc[]
```
Comment (michaelryanmcneill) [inline modules/configuring-a-proxy-after-installation-ocm.adoc:45]: Suggested to remove the snippet.

File: modules/osd-create-cluster-ccs.adoc [hunk 4]
```diff
@@ -304,15 +304,17 @@ For more information, see the requirements for _Security groups_ under _Addition
...
 .. Enter a value in at least one of the following fields:
 ** Specify a valid *HTTP proxy URL*.
 ** Specify a valid *HTTPS proxy URL*.
 ** In the *Additional trust bundle* field, provide a PEM encoded X.509 certificate bundle. The bundle is added to the trusted certificate store for the cluster nodes. An additional trust bundle file is required unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle.
 +
 If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional certificate authorities (CAs), you must provide the MITM CA certificate.
-.. Click *Next*.
++
 --
+include::snippets/rosa-add-trust-bundle-policy.adoc[]
```
Comment (michaelryanmcneill) [inline modules/osd-create-cluster-ccs.adoc:315]: Suggested to remove the snippet.

File: modules/osd-create-cluster-gcp-account.adoc [hunk 5]
```diff
@@ -127,20 +127,17 @@ If you are installing a cluster into a Shared VPC, the VPC name and subnets are
...
 +
 If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional certificate authorities (CAs), you must provide the MITM CA certificate.
 +
-[NOTE]
-====
-If you upload an additional trust bundle file without specifying an HTTP or HTTPS proxy URL, the bundle is set on the cluster but is not configured to be used with the proxy.
-====
-.. Click *Next*.
 --
+include::snippets/rosa-add-trust-bundle-policy.adoc[]
```
Comment (michaelryanmcneill) [inline modules/osd-create-cluster-gcp-account.adoc:138]: Suggested to remove the snippet.

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 6]
```diff
@@ -126,20 +126,17 @@ If you are installing a cluster into a Shared VPC, the VPC name and subnets are
...
 +
 If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional certificate authorities (CAs), you must provide the MITM CA certificate.
 +
-[NOTE]
-====
-If you upload an additional trust bundle file without specifying an HTTP or HTTPS proxy URL, the bundle is set on the cluster but is not configured to be used with the proxy.
-====
-.. Click *Next*.
 --
+include::snippets/rosa-add-trust-bundle-policy.adoc[]
```
Comment (michaelryanmcneill) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:137]: Suggested to remove the snippet.

File: modules/rosa-sts-creating-a-cluster-with-customizations-ocm.adoc [hunk 7]
```diff
@@ -318,20 +318,17 @@ For more information, see the requirements for _Security groups_ under _Addition
...
 +
 If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional certificate authorities (CAs), you must provide the MITM CA certificate.
 +
-[NOTE]
-====
-If you upload an additional trust bundle file without specifying an HTTP or HTTPS proxy URL, the bundle is set on the cluster but is not configured to be used with the proxy.
-====
-.. Click *Next*.
 --
+include::snippets/rosa-add-trust-bundle-policy.adoc[]
```
Comment (michaelryanmcneill) [inline modules/rosa-sts-creating-a-cluster-with-customizations-ocm.adoc:329]: Suggested to remove the snippet.

File: snippets/rosa-add-trust-bundle-policy.adoc [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+//This snippet appears in the following modules:
+//
+// * ../rosa-configuring-a-proxy-after-installation-ocm.adoc
+// * ../rosa-sts-creating-a-cluster-with-customizations-ocm.adoc
+// * ../osd-create-cluster-gcp-account.adoc
+// * ../osd-create-cluster-rhm-gcp-account.adoc
+// * ../osd-create-cluster-ccs.adoc
+
+:_mod-docs-content-type: SNIPPET
+[NOTE]
```
Comment (michaelryanmcneill) [inline snippets/rosa-add-trust-bundle-policy.adoc:18]: Suggested to remove the snippet entirely.

File: modules/rosa-sts-creating-a-cluster-with-customizations-ocm.adoc [hunk 9]
```diff
@@ -318,20 +318,17 @@ For more information, see the requirements for _Security groups_ under _Addition
 
 . If you opted to configure a cluster-wide proxy, provide your proxy configuration details on the *Cluster-wide proxy* page:
 +
---
 .. Enter a value in at least one of the following fields:
 ** Specify a valid *HTTP proxy URL*.
 ** Specify a valid *HTTPS proxy URL*.
 ** In the *Additional trust bundle* field, provide a PEM encoded X.509 certificate bundle. The bundle is added to the trusted certificate store for the cluster nodes. An additional trust bundle file is required unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle.
```
Comment (michaelryanmcneill) [inline modules/rosa-sts-creating-a-cluster-with-customizations-ocm.adoc:324]: Suggested revision:

> In the *Additional trust bundle* field, provide a PEM encoded X.509 certificate bundle. The bundle is added to the trusted certificate store for the cluster nodes. An additional trust bundle file is required for users who use a TLS-inspecting proxy unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle. This applies regardless of whether the proxy is transparent or requires explicit configuration using the `http-proxy` and `https-proxy` arguments.

File: modules/rosa-sts-creating-a-cluster-with-customizations-ocm.adoc [hunk 10]
```diff
@@ -318,20 +318,17 @@ For more information, see the requirements for _Security groups_ under _Addition
...
 +
 If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional certificate authorities (CAs), you must provide the MITM CA certificate.
```
Comment (michaelryanmcneill) [inline modules/rosa-sts-creating-a-cluster-with-customizations-ocm.adoc:326]: Suggested to remove this line.

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 11]
```diff
@@ -126,20 +126,17 @@ If you are installing a cluster into a Shared VPC, the VPC name and subnets are
 . Click *Next*.
 . If you opted to configure a cluster-wide proxy, provide your proxy configuration details on the *Cluster-wide proxy* page:
 +
---
 .. Enter a value in at least one of the following fields:
 ** Specify a valid *HTTP proxy URL*.
 ** Specify a valid *HTTPS proxy URL*.
 ** In the *Additional trust bundle* field, provide a PEM encoded X.509 certificate bundle. The bundle is added to the trusted certificate store for the cluster nodes. An additional trust bundle file is required unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle.
```
Comment (michaelryanmcneill) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:132]: Suggested revision:

> In the *Additional trust bundle* field, provide a PEM encoded X.509 certificate bundle. The bundle is added to the trusted certificate store for the cluster nodes. An additional trust bundle file is required for users who use a TLS-inspecting proxy unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle. This applies regardless of whether the proxy is transparent or requires explicit configuration using the `http-proxy` and `https-proxy` arguments.

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 12]
```diff
@@ -126,20 +126,17 @@ If you are installing a cluster into a Shared VPC, the VPC name and subnets are
...
 +
 If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional certificate authorities (CAs), you must provide the MITM CA certificate.
```
Comment (michaelryanmcneill) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:134]: Suggested to remove this line.

File: modules/osd-create-cluster-gcp-account.adoc [hunk 13]
```diff
@@ -127,20 +127,17 @@ If you are installing a cluster into a Shared VPC, the VPC name and subnets are
 . Click *Next*.
 . If you opted to configure a cluster-wide proxy, provide your proxy configuration details on the *Cluster-wide proxy* page:
 +
---
 .. Enter a value in at least one of the following fields:
 ** Specify a valid *HTTP proxy URL*.
 ** Specify a valid *HTTPS proxy URL*.
 ** In the *Additional trust bundle* field, provide a PEM encoded X.509 certificate bundle. The bundle is added to the trusted certificate store for the cluster nodes. An additional trust bundle file is required unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle.
```
Comment (michaelryanmcneill) [inline modules/osd-create-cluster-gcp-account.adoc:133]: Suggested revision:

> In the *Additional trust bundle* field, provide a PEM encoded X.509 certificate bundle. The bundle is added to the trusted certificate store for the cluster nodes. An additional trust bundle file is required for users who use a TLS-inspecting proxy unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle. This applies regardless of whether the proxy is transparent or requires explicit configuration using the `http-proxy` and `https-proxy` arguments.

File: modules/osd-create-cluster-gcp-account.adoc [hunk 14]
```diff
@@ -127,20 +127,17 @@ If you are installing a cluster into a Shared VPC, the VPC name and subnets are
...
 +
 If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional certificate authorities (CAs), you must provide the MITM CA certificate.
```
Comment (michaelryanmcneill) [inline modules/osd-create-cluster-gcp-account.adoc:135]: Suggested to remove this line.

File: modules/osd-create-cluster-ccs.adoc [hunk 15]
```diff
@@ -304,15 +304,17 @@ For more information, see the requirements for _Security groups_ under _Addition
 endif::osd-on-aws[]
 . If you opted to configure a cluster-wide proxy, provide your proxy configuration details on the *Cluster-wide proxy* page:
 +
---
 .. Enter a value in at least one of the following fields:
 ** Specify a valid *HTTP proxy URL*.
 ** Specify a valid *HTTPS proxy URL*.
 ** In the *Additional trust bundle* field, provide a PEM encoded X.509 certificate bundle. The bundle is added to the trusted certificate store for the cluster nodes. An additional trust bundle file is required unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle.
```
Comment (michaelryanmcneill) [inline modules/osd-create-cluster-ccs.adoc:310]: Suggested revision:

> In the *Additional trust bundle* field, provide a PEM encoded X.509 certificate bundle. The bundle is added to the trusted certificate store for the cluster nodes. An additional trust bundle file is required for users who use a TLS-inspecting proxy unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle. This applies regardless of whether the proxy is transparent or requires explicit configuration using the `http-proxy` and `https-proxy` arguments.

File: modules/osd-create-cluster-ccs.adoc [hunk 16]
```diff
@@ -304,15 +304,17 @@ For more information, see the requirements for _Security groups_ under _Addition
...
 +
 If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional certificate authorities (CAs), you must provide the MITM CA certificate.
```
Comment (michaelryanmcneill) [inline modules/osd-create-cluster-ccs.adoc:312]: Suggested to remove this line.

File: modules/configuring-a-proxy-after-installation-ocm.adoc [hunk 17]
```diff
@@ -41,6 +41,9 @@ endif::openshift-dedicated[]
 +
```
Comment (michaelryanmcneill) [inline modules/configuring-a-proxy-after-installation-ocm.adoc:41]: Suggested revision:

> In the *Additional trust bundle* field, provide a PEM encoded X.509 certificate bundle. The bundle is added to the trusted certificate store for the cluster nodes. An additional trust bundle file is required for users who use a TLS-inspecting proxy unless the identity certificate for the proxy is signed by an authority from the {op-system-first} trust bundle. This applies regardless of whether the proxy is transparent or requires explicit configuration using the `http-proxy` and `https-proxy` arguments.

File: modules/configuring-a-proxy-after-installation-ocm.adoc [hunk 18]
```diff
@@ -41,6 +41,9 @@ endif::openshift-dedicated[]
...
 If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional certificate authorities (CAs), you must provide the MITM CA certificate.
```
Comment (michaelryanmcneill) [inline modules/configuring-a-proxy-after-installation-ocm.adoc:42]: Suggested to remove this line.

**Review by michaelryanmcneill — CHANGES_REQUESTED** [review]
Updated with additional suggestions on content changes.

File: modules/configuring-a-proxy-during-installation-cli.adoc (hunk 2 above)
Comment (AedinC) [reply modules/configuring-a-proxy-during-installation-cli.adoc:33]: Thanks @michaelryanmcneill, do I still keep in this line:  
`If you use an MITM transparent proxy network that does not require additional proxy configuration but requires additional certificate authorities (CAs), you must provide the MITM CA certificate.`? Thanks.

Comment (michaelryanmcneill) [reply modules/configuring-a-proxy-during-installation-cli.adoc]: @AedinC Yes, that line should be removed.

**Comment by michaelryanmcneill** [comment]
@AedinC While we're updating this document, can we change this wording to make more sense:

> NOTE
> When using a cluster-wide proxy, you must configure the s3.<aws_region>.amazonaws.com endpoint as type Gateway. Also, you can configure the ec2.<aws_region>.amazonaws.com and elasticloadbalancing.<aws_region>.amazonaws.com endpoints only as type Interface.

I'd propose the following:
> NOTE
> When using a cluster-wide proxy, you must configure the `s3.<aws_region>.amazonaws.com` endpoint as type `Gateway`. 

The EC2 and ELB endpoints do not offer gateway endpoints and the way it is worded is quite confusing anyways.

**Comment by AedinC** [comment]
@bchandra-ocp I have added feedback from Michael. Can you please take a look and let me know if you want anything added? Thanks.

**Pull request merged**